### ISSUE TITLE
k8s version matrix

### ISSUE THREAD

**Issue by kuritka** [k8gb-io/k8gb#1044]
We need to verify that k8gb will run on different versions of kubernetes. We could take one simple terratest and run it on several k3d-actions but with different k3s versions.

We are already doing something similar when we test our libraries across different GO versions: [here](https://github.com/k8gb-io/go-weight-shuffling/blob/main/.github/workflows/test.yaml#L18-L20)

**Comment by ytsarev** [comment]
@kuritka while supporting k8s version matrix totally makes sense, do we really need to keep compatibility with older golang versions? what is the motivation to do so? we are just shipping controller binary, not a library so maybe we can simplify the golang test matrix?

**Comment by kuritka** [comment]
Hi @ytsarev , sorry, it's a misunderstanding. I mean the matrix for k8s version only. GO versions were  given as an example of how it should work. 

thx

**Comment by Horiodino** [comment]
Hii  @kuritka @ytsarev  i would like to contribute to this project but can you guys give more context about this issue.

**Comment by ytsarev** [comment]
Hi @Horiodino , welcome to the project! We really appreciate your willingness to contribute! 

The context is pretty simple, it is about to provider well-tested version of https://github.com/k8gb-io/k8gb#production-readiness table supported by tests.

**Comment by Horiodino** [comment]
> Hi @Horiodino , welcome to the project! We really appreciate your willingness to contribute!
> 
> The context is pretty simple. It is about to provide a well-tested version of https://github.com/k8gb-io/k8gb#production-readiness table supported by automated tests.

than-you @ytsarev  for replying , so basically it for testing the k8sgb on different kubernetes version and go lang for providing more compatibility  ?

**Comment by kuritka** [comment]
Hi @Horiodino , appreciate your interresets. Doing Golang version matrix is not important in case of k8gb project and would complicate the pipeline unnecessarily. I just used it as an example of how to implement a matrix and I didn't have anything better at hand at the time. Different versions of kubernetes are important.

**Comment by ytsarev** [comment]
@itsfarhan feel free to pick this one up if it looks interesting :)

**Comment by itsfarhan** [comment]
@ytsarev is this similar to #1909 ?

**Comment by ytsarev** [comment]
@itsfarhan good point, i think we can consolidate on #1909 that is currently in the works. Closing this one, thanks!
